src/ssl.mli

```diff
@@ -523,3 +514,40 @@ val input_int : socket -> int
 
 (** Write an integer on an SSL socket. *)
 val output_int : socket -> int -> unit
+
+(** In `SslNoRelease`, you will find equivalent of the previous function that
+    do not release the OCaml runtime lock before calling the underlying SSL
+    primitives. This should be preferer for non blocking sockets where the
+    usual semantics apply, i.e. handling of `EWOULDBLOCK`, `EGAIN`, etc.
+    In could also be used with blocking socket if you program has only one thread.
+
+    This saves some buffer copy and can be up to 2x faster.
+*)
+module SslNoRelease : sig
+  val connect : socket -> unit
```

Comment:
it shouldn't be too much work to copy the docstrings here too, and it'll provide a better experience for users.

Reply:
> it shouldn't be too much work to copy the docstrings here too, and it'll provide a better experience for users.

Copying means the user should read both to see there is no diff, no copying means you are sure it is the same documentation ? It also keeps the size of mli minimal ?

Reply:
Copying the docstring inline means the users will be able to see the docs for every function from their editor integration.

Reply:
I forgot about editor integration! I don't use it. I guess exporting the module type would solve that? There is no [@@ same doc as ...] which could be nice. I think a bit more and probably will copy the docstring.

Reply:
Fixed by [358177a](https://github.com/savonet/ocaml-ssl/pull/106/commits/358177a7952dbede5f6bec34abeab735cf4f8a36)